openmdao/core/component.py

```diff
@@ -1702,10 +1702,18 @@ def _check_consistent_serial_dinputs(self, nz_dist_outputs):
 
         if self._serial_idxs is None:
             ranges = defaultdict(list)
+            output_len = 0
+            for name, offset, end, vec, slc, dist_sizes in self._jac_wrt_iter():
+                if vec is self._outputs:
+                    output_len = len(self._outputs)
+                    break
```

Comment:
Ugh, sorry my first recommended fix was wrong.  Since an explicit comp will have all inputs and an implicit comp will have all outputs followed by all inputs, how about this:

```
            output_len = len(self._outputs) if self.is_explicit() else 0
            for name, offset, end, vec, slc, dist_sizes in self._jac_wrt_iter():
                if dist_sizes is None:  # not distributed
                    if offset != end:
                        if vec is self._outputs:
                            ranges[vec].append(range(offset, end))
                        else:
                            ranges[vec].append(range(offset - output_len, end - output_len))

```